Makefile

```diff
@@ -0,0 +1,163 @@
+ifneq ($(BUILD_TAG), latest)
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) || true
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		backend-services/$@/
+ifeq ($(PUSH), true)
+	@docker tag artemis-$@:$(BUILD_TAG) $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: $(TAP_SERVICES)
+$(TAP_SERVICES): # build tap container
+$(TAP_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest || true
+ifneq ($(BUILD_TAG), latest)
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) || true
+endif
+	@docker build -t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) \
+		monitor-services/$@/
+ifeq ($(PUSH), true)
+	@docker tag artemis-$@:$(BUILD_TAG) $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+	@docker push $(CONTAINER_REPO)/artemis-$@:${BUILD_TAG}
+endif
+
+.PHONY: $(FRONTEND_SERVICES)
+$(FRONTEND_SERVICES): # build frontend container
+$(FRONTEND_SERVICES):
+	@echo "Building $@ service for tag $(BUILD_TAG)"
+	@docker pull $(CONTAINER_REPO)/artemis-$@:latest || true
+ifneq ($(BUILD_TAG), latest)
+	@docker pull $(CONTAINER_REPO)/artemis-$@:$(BUILD_TAG) || true
+endif
+	@docker build --build-arg revision=$(git rev-parse --short HEAD) \
+		-t artemis-$@:$(BUILD_TAG) \
+		--cache-from $(CONTAINER_REPO)/artemis-frontend:latest \
+		--cache-from $(CONTAINER_REPO)/artemis-frontend:$(BUILD_TAG) frontend/
+ifeq ($(PUSH), true)
+	@docker tag artemis-$@:$(BUILD_TAG) $(CONTAINER_REPO)/artemis-temp$@:${BUILD_TAG}
+	@docker push $(CONTAINER_REPO)/artemis-temp$@:${BUILD_TAG}
+endif
+
+.PHONY: build-backend
+build-backend: # builds all backend containers
+build-backend: $(BACKEND_SERVICES)
+
+.PHONY: build-taps
+build-taps: # builds all tap containers
+build-taps: $(TAP_SERVICES)
+
+.PHONY: build-frontend
+build-frontend: # builds frontend container
+build-frontend: $(FRONTEND_SERVICES)
+
+.PHONY: migration-check
+migration-check: # checks if migration is not broken
+migration-check:
+	@wget -q -O postgres-data-current.tar.gz --no-check-certificate 'https://docs.google.com/uc?export=download&id=1UwtIp7gF5uO5PfhTbOAMPpDLJ4H55m7a'
+	@tar xzf postgres-data-current.tar.gz
+	@docker-compose up -d postgres
+	@sleep 10
+	@docker-compose logs postgres | grep "database system is ready to accept connections"
+	@docker-compose down -v --remove-orphans && sudo rm -rf postgres-data-current*
+
+.PHONY: verify-configuration
+verify-configuration: # verify that configuration variables are changed correctly
+verify-configuration:
+	@python other/verify_script.py
+
+.PHONE: setup-dev
+setup-dev: # pull all images and tag them for local development
+setup-dev:
+	@for service in $(SERVICES) ; do \
+		docker-compose pull $$service; \
```

Comment:
what happens here though if image is non-existent? This is why travis fails the tests if I get this correctly.